submittal/src/main/java/com/kyriba/submittal/api/SpeakerController.java

```diff
@@ -0,0 +1,104 @@
+package com.kyriba.submittal.api;
+
+import com.kyriba.submittal.domain.dto.PersonalInfoDto;
+import com.kyriba.submittal.domain.dto.SpeakerDataDto;
+import com.kyriba.submittal.domain.dto.TopicDto;
+import com.kyriba.submittal.service.SpeakerService;
+import lombok.Builder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.SessionAttribute;
+
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
+
+
+/**
+ * @author M-ABL
+ */
+// TODO: setup security. Let's assume a filter retrieves a speaker's id by email/password and puts the id to the session.
+@RestController
+@RequestMapping(value = "/api/v1/speaker",
+    consumes = APPLICATION_JSON_UTF8_VALUE,
+    produces = APPLICATION_JSON_UTF8_VALUE)
+public class SpeakerController
+{
+  private final SpeakerService speakerService;
+
+
+  @Autowired
+  SpeakerController(final SpeakerService speakerService)
+  {
+    this.speakerService = speakerService;
+  }
+
+
+  @PostMapping(path = "/register")
+  public SpeakerResponse register(@RequestBody final SpeakerRequest speaker)
+  {
+    // TODO: can the controller invoke Notification MS directly in order to send a confirmation email?
+    String key = speakerService.register(
+        speaker.email, speaker.password, speaker.personalData, speaker.topics, speaker.conferenceId);
+    return SpeakerResponse.builder()
+        .email(speaker.email)
+        .confirmationKey(key)
+        .build();
+  }
+
+
+  @PostMapping(path = "/topics")
+  public TopicResponse proposeTopics(@SessionAttribute("speakerId") long speakerId,

```

Comment:
On POST some kind of resource ID is returned

Reply:
TopicResponse contains id of a speaker and id of each created topic. What else I missed?

Reply:
It is expected to return only ID of just created Topic.